subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -330,18 +329,20 @@ static void isr_tx(void *param)
 	}
 	/* TODO: MOVE ^^ */
 
-	radio_isr_set(isr_rx, param);
+	/* setup tIFS switching */
 	radio_tmr_tifs_set(EVENT_IFS_US);
 	radio_switch_complete_and_tx(0, 0, 0, 0);
-	radio_pkt_rx_set(radio_pkt_scratch_get());
 
+	radio_pkt_rx_set(radio_pkt_scratch_get());
 	/* assert if radio packet ptr is not set and radio started rx */
 	LL_ASSERT(!radio_is_ready());
 
 	if (IS_ENABLED(CONFIG_BT_CTLR_PROFILE_ISR)) {
 		lll_prof_cputime_capture();
 	}
 
+	radio_isr_set(isr_rx, param);
```

Comment:
@cvinayak Moving radio_isr_set() after radio_switch_complete_and_tx() makes any difference?

Reply:
Couple CPU cycles, yes.

Reply:
@cvinayak  But is it related with the 4us you removed?

Reply:
@George-Stefan 4us removed is in separate commit and refactoring to move non critical code later in the ISR is another commit.